File: Python/8py 1/correct.py
"""
|--------------------------------------------------------------------------
| Correct the mistakes of the character recognition software
|--------------------------------------------------------------------------
|
"""

def correct(s):
   return s.replace('5', 'S').replace('0', 'O').replace('1', 'I')


def correct(string):
   for e, c in {'5':'S','0':'O','1':'I'}.items():
      string = string.replace(e,c)
   return string


def correct(string):
   return string.translate(str.maketrans("501", "SOI"))


def correct(string):
   return ''.join({'0':'O', '5':'S', '1':'I'}.get(c, c) for c in string)


print(correct("L0ND0N"), "LONDON")
print(correct("DUBL1N"), "DUBLIN")
print(correct("51NGAP0RE"), "SINGAPORE")
print(correct("BUDAPE5T"), "BUDAPEST")
print(correct("PAR15"), "PARIS")

File: JavaScript/8js 1/correct.js
/*
|--------------------------------------------------------------------------
| Correct the mistakes of the character recognition software
|--------------------------------------------------------------------------
|
*/

function correct(string) {
   let res = '';
   for (let i = 0; i < string.length; i++) {
      res += string[i] == '5' ? 'S' : string[i] == '0' ? 'O' : string[i] == '1' ? 'I' : string[i];
   }
   return res;
}


function correct(string) {
   return string.replace(/0/g, "O")
      .replace(/5/g, "S")
      .replace(/1/g, "I");
}


function correct(string) {
   return [...string].map(a => ({
      '0': 'O',
      '5': 'S',
      '1': 'I'
   })[a] || a).join('')
}


function correct(string) {
   return string.replace(/\d/g, val => `OI   S` [val]);
}


console.log(correct("L0ND0N"), "LONDON");
console.log(correct("DUBL1N"), "DUBLIN");
console.log(correct("51NGAP0RE"), "SINGAPORE");
console.log(correct("BUDAPE5T"), "BUDAPEST");
console.log(correct("PAR15"), "PARIS");